File: Design-Patterns-Using-JavaScript/AntiPattern/Spaghetti.js
class Spaghetti {
    constructor() {
        // Define connection parameters
        this.url = 'jdbc:postgresql://localhost:5432/mydb';
        this.user = 'user';
        this.password = 'password';
    }

    async processData(data) {
        // Validate input
        if (!data || data.trim() === '') {
            console.log('Invalid data');
            return null;
        }

        // Initialize variables
        let client = null;
        let resultSet = null;

        try {
            // Connect to the database
            client = new Client({
                connectionString: this.url,
                user: this.user,
                password: this.password
            });
            await client.connect();

            // Process data
            const query = 'SELECT * FROM mytable WHERE data = $1';
            const values = [data];
            resultSet = await client.query(query, values);

            return resultSet.rows;
        } catch (error) {
            console.error('Error executing query:', error);
            return null;
        } finally {
            // Close the client connection
            if (client) {
                await client.end();
            }
        }
    }
}

// Client code
const spaghetti = new Spaghetti();
const data = 'exampleData';
spaghetti.processData(data);

File: Design-Patterns-Using-Cpp/AntiPattern/Spaghetti.cpp
#include <iostream>
#include <postgresql/libpq-fe.h>

class Spaghetti {
public:
    static PGresult* processData(std::string data) {
        // Validate input
        if (data.empty()) {
            std::cout << "Invalid data" << std::endl;
            return nullptr;
        }

        // Define connection parameters
        const char* conninfo = "host=localhost port=5432 dbname=mydb user=user password=password";

        // Initialize variables
        PGconn *conn = nullptr;
        PGresult *res = nullptr;

        // Connect to the database
        conn = PQconnectdb(conninfo);

        // Check connection status
        if (PQstatus(conn) != CONNECTION_OK) {
            std::cerr << "Connection to database failed: " << PQerrorMessage(conn) << std::endl;
            PQfinish(conn);
            return nullptr;
        }

        // Process data
        const char* paramValues[1] = {data.c_str()};
        res = PQexecParams(conn, "SELECT * FROM mytable WHERE data = $1",
                            1,       /* number of parameters */
                            nullptr, /* let the backend deduce param type */
                            paramValues,
                            nullptr, nullptr, 0);

        if (PQresultStatus(res) != PGRES_TUPLES_OK) {
            std::cerr << "Query execution failed: " << PQerrorMessage(conn) << std::endl;
            PQclear(res);
            PQfinish(conn);
            return nullptr;
        }

        return res;
    }

    // Example usage
    int main() {
        std::string data = "exampleData";
        PGresult *results = processData(data);
        // Process the results if needed
        PQclear(results); // Free the result memory
        return 0;
    }
};
